doc/source/container_builder.rst

```diff
@@ -0,0 +1,276 @@
+btrfs as the storage backend. The following is only required if your
+host system root filesystem is not btrfs.
+
+.. code:: bash
+
+    $ sudo qemu-img create /var/lib/docker-storage.btrfs 20g
+    $ sudo mkfs.btrfs /var/lib/docker-storage.btrfs
+    $ sudo mkdir -p /var/lib/docker
+    $ sudo mount /var/lib/docker-storage.btrfs /var/lib/docker
+
+    $ sudo vi /etc/fstab
+
+      /var/lib/docker-storage.btrfs /var/lib/docker btrfs defaults 0 0
+
+    $ sudo vi /etc/sysconfig/docker
+
+      DOCKER_OPTS="-s btrfs"
+
+Finally start the docker daemon:
+
+.. code:: bash
+
+    $ sudo systemctl restart docker
+
+Installing and Setting up the build container
+----------------------------------------------
+
+In order to build in a contained environment docker has to start a
+privileged system container. Such a container must be imported before
+docker can use it. The build container is provided to you as a
+service and build with KIWI in the project
+at https://build.opensuse.org/project/show/Virtualization:Appliances:Images
+On a regular basis the result image is pushed
+to https://hub.docker.com/r/schaefi/kiwi-build-box
+
+There are two ways to import the build container to your local docker system
+
+1. Download from the openSUSE Buildservice and manually import
+2. Use docker to pull the box from dockerhub
+
+Pull from dockerhub
+-------------------
+
+.. code:: bash
+
+    $ docker pull schaefi/kiwi-build-box:latest
+
+Download from the open BuildService
+-----------------------------------
+
+Download the .tar.bz2 file which starts with :file:`Docker-Tumbleweed`
+
+.. code:: bash
+
+    $ wget http://download.opensuse.org/repositories/Virtualization:/Appliances:/Images/images/Docker-Tumbleweed.XXXXXXX.docker.tar.xz
+
+Import the downloaded tarball to docker as follows:
+
+.. code:: bash
+
+    $ cat Docker-Tumbleweed.XXXXXXX.docker.tar.xz | docker import - schaefi/kiwi-build-box:latest
+
+
+Installing and Setting up vagrant
+---------------------------------
+
+.. note:: Optional step
+
+    This step can be skipped if there are no complex provision tasks
+    of the building environment required.
+
+Installing vagrant is well documented at
+https://docs.vagrantup.com/v2/installation/index.html
+
+Access to a machine started by vagrant is done through ssh exclusively.
+Because of that an initial key setup is required in the box vagrant should
+start. The kiwi provided build boxes includes the public key of the vagrant
+key pair and thus allows access. It is important to understand that the
+private vagrant key is not a secure key because the private key is not
+protected. However this is not a problem because vagrant creates a new
+key pair for each machine it starts. In order to allow vagrant the initial
+access and the creation of a new key pair, it's required to provide access
+to the insecure vagrant private key. The following commands should not be
+executed as root, but rather as the user intending to build images.
```

Comment:
"as the user intending to build images." -> "the intended user to build images."